docs/cvl/hooks.md

```diff
@@ -361,6 +361,32 @@ hook SELFDESTRUCT(address a)
 % The standard stack-manipulating instructions `DUP*`, `SWAP*`, `PUSH*` and `POP`
 % are not modeled.  `MLOAD` and `MSTORE` are also not modeled.
 
+(call-hooks)=
+### Call hooks
+
+We provide hooks for all four call opcodes: [`CALL`](https://www.evm.codes/#f1),
+[`CALLCODE`](https://www.evm.codes/#f2), [`DELEGATECALL`](https://www.evm.codes/#f4),
+and [`STATICCALL`](https://www.evm.codes/#fa).
+The hook parameters match the stack inputs of the respective opcodes.
+
+The arguments for the call arguments and return values (`argsOffset`, `argsSize`,
+`retOffset`, and `retSize`) only exist for future use, they can not be used to
+access the respective data right now.
+
+These hooks can be very useful to establish sensible security invariants.
+For example, both `CALLCODE` and `DELEGATECALL` can call external code on the
+current context which exposes private data to the called external code:
```

Comment:
maybe explain a little more why you wrote each hook the way you did. e.g. why limit in callcode only to calls within the `currentContract` (i.e., it's the only one we're interested in checking),
and for the delegatecall it should also be elaborated - either we are not in the contract of interest, or that contract (or another) is calling into itself.
(btw it will likely have tons of false positives in many types of code that link to trusted 3rd party libraries with external functions, so it's quite specific)

Reply:
I added some reasoning that makes this clearer.